### PR TITLE
feat(nav): style drilldown menu for nav

### LINKED ISSUE #6919: Nav - drilldown needs to be updated for v6
This is what drilldown looks like currently - https://patternfly-react-v6.surge.sh/components/navigation/react-demos/drilldown-nav/

Screenshot 2024-07-24 at 1 42 29 PM

cc @lboehling @andrew-ronaldson

### PR DESCRIPTION
**Pull request by srambach**
Fixes #6919

### CODE REVIEW

**Review by andrew-ronaldson — APPROVED** [review]
fire away

File: src/patternfly/components/Nav/nav.scss [hunk 1]
```diff
@@ -135,9 +135,36 @@
...
+        gap: var(--#{$nav}__subnav--RowGap);
+      }
+    
+      .#{$menu}__item {
+        border-radius: var(--#{$nav}__link--BorderRadius);
+    
+        &:hover,
+        &:focus {
+          --#{$menu}__item--BackgroundColor: var(--#{$nav}__link--hover--BackgroundColor);
+        }
```
Comment (mcoker) [inline src/patternfly/components/Nav/nav.scss:163]: Curious if we could remove this and get the same with `--pf-v6-c-menu__list-item--hover--BackgroundColor: var(--pf-v6-c-nav__link--hover--BackgroundColor);` in the `&.pf-m-drilldown` above? We would then want to update the `border-radius` so it goes on `.pf-v6-c-menu__list-item` and `.pf-v6-c-menu__list-item::before` (or maybe just `.pf-v6-c-menu__list-item::before` if that works?) and clean up the existing item/list-item background vars on `&.pf-m-drilldown`

File: src/patternfly/components/Nav/nav.scss [hunk 2]
```diff
@@ -135,9 +135,36 @@
...
+      --#{$menu}--BackgroundColor: var(--#{$nav}--BackgroundColor);
+      --#{$menu}--BoxShadow: none;
+      --#{$menu}__list-item--Color: var(--#{$nav}__link--Color);
+      --#{$menu}__list-item--BackgroundColor: transparent;
+      --#{$menu}__list-item--hover--BackgroundColor: var(--#{$nav}__link--hover--BackgroundColor);
+      --#{$menu}__list-item--hover--BackgroundColor: transparent;
+      --#{$menu}__item--BackgroundColor: var(--#{$nav}__link--BackgroundColor);
+
+      .#{$menu}__list {
+        gap: var(--#{$nav}__subnav--RowGap);
```
Comment (mcoker) [inline src/patternfly/components/Nav/nav.scss:154]: Just so we don't need to change this if we ended up adding a column-gap for some reason

```suggestion
        row-gap: var(--#{$nav}__subnav--RowGap);
```

**Review by mcoker — COMMENTED** [review]
Nice! LGTM, just some nit comments.

**Review by mcoker — APPROVED** [review]
LGTM! 🥳

**Pull request merged**